File: python/LeetCode_374_GuessNumberHigherorLower.py
def guessNumber(self, n: int) -> int:
    low = 0
    high = n
    res = guess(n) 
    while (res != 0):
        mid = low + ((high - low) // 2)

        res = guess(mid)  
        if (res == 0):
            return mid
        elif (res == 1):
            low = mid
        elif (res == -1):
            high = mid

    return n

File: c/LeetCode_374_GuessNumberHigherorLower.c
#include <stdio.h>
#include <stdlib.h>


int guessNumber(int n){
    int low = 0;
    int high = n;
    int res = guess(n);
    while (res != 0) {
        int mid = low + ((high - low) / 2);

        res = guess(mid);
        if (res == 0) {
            return mid;
        } else if (res == 1) {
            low = mid;
        } else if (res == -1) {
            high = mid;
        }
    }

    return n;
}


int main() {

    return 0;
}